### PR TITLE
[EDGNCIP-26/EDGNCIP-27] - Enhance HTTP Endpoint Security with TLS and FIPS-140-2 Compliant Cryptography

### PR DESCRIPTION
**Pull request by azizbekxm**
https://folio-org.atlassian.net/browse/EDGNCIP-27
WebServer creation occurs in edge-common, so additional changes in particluar module for this story is not needed.

https://folio-org.atlassian.net/browse/EDGNCIP-26
OrdersOkapiClinentFactory is responsible for creating WebClients with SSL or without SSL support based on system properties that is implemented in scope of this story

### CODE REVIEW

File: src/main/java/org/folio/edge/ncip/NcipHandler.java [hunk 1]
```diff
@@ -1,156 +1,117 @@
...
-			      ErrorMessage error = new ErrorMessage(statusCode, message);
-			      finalMsg = error.toJson();
-			    }
-			    catch(JsonProcessingException ex){
-			      finalMsg = "{ code : \"\", message : \"" + message + "\" }";
-			    }
-			    return finalMsg;
-		}
-
-		  protected void handleProxyResponse(RoutingContext ctx, HttpClientResponse resp) {
```
Comment (azizbekxm) [inline src/main/java/org/folio/edge/ncip/NcipHandler.java:127]: It is not used so, it is deleted

Comment (SerhiiNosko) [reply src/main/java/org/folio/edge/ncip/NcipHandler.java:127]: I see that it is used in handleHealthCheck()

Comment (azizbekxm) [reply src/main/java/org/folio/edge/ncip/NcipHandler.java:127]: Thanks, reverting

File: src/main/java/org/folio/edge/ncip/utils/NcipOkapiClient.java [hunk 2]
```diff
@@ -1,69 +1,58 @@
...
-	  }
+  public NcipOkapiClient(OkapiClient client) {
+    super(client);
+  }
+
+  public void ncipConfigCheck(String payload, MultiMap headers,
+                              Handler<HttpResponse<Buffer>> responseHandler,
+                              Handler<Throwable> exceptionHandler) {
+
+    System.out.println("in send get request in the client");
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/edge/ncip/utils/NcipOkapiClient.java:20]: if you already started formatting, so can change thse lines to use logger

File: src/test/java/org/folio/edge/ncip/MainVerticleTest.java [hunk 3]
```diff
@@ -151,41 +146,36 @@ public void testConfigCheck(TestContext context) throws Exception {
...
   public void testErrorMessage() {
 	  ErrorMessage errorMessage = new ErrorMessage(200,"ok");
-	  assertFalse(errorMessage.equals("ok"));
+    assertEquals("ok", errorMessage.message);
   }
   
   @Test
   public void testErrorMessagesEquality() {
 	  ErrorMessage errorMessageOrig = new ErrorMessage(500,"bad request");
 	  ErrorMessage errorMessage = new ErrorMessage(200,"ok");
```
Comment (SerhiiNosko) [inline src/test/java/org/folio/edge/ncip/MainVerticleTest.java:159]: if you formatted this file so these 2 lines with error messages also should be formatted

File: src/main/java/org/folio/edge/ncip/NcipHandler.java (hunk 1 above)
Comment (SerhiiNosko) [reply src/main/java/org/folio/edge/ncip/NcipHandler.java:127]: Yes it was not used, instead org.folio.edge.core.Handler::handleProxyResponse is used

**Pull request closed without merging**